internal/v4/common_test.go

```diff
@@ -291,6 +293,91 @@ func (s *commonSuite) addRequiredCharms(c *gc.C, bundle charm.Bundle) {
 	}
 }
 
+func (s *commonSuite) assertPut(c *gc.C, url string, val interface{}) {
+	s.assertPut0(c, url, val, false)
+}
+
+func (s *commonSuite) assertPutAsAdmin(c *gc.C, url string, val interface{}) {
+	s.assertPut0(c, url, val, true)
+}
+
+func (s *commonSuite) assertPut0(c *gc.C, url string, val interface{}, asAdmin bool) {
+	body, err := json.Marshal(val)
+	c.Assert(err, gc.IsNil)
+	p := httptesting.JSONCallParams{
+		Handler: s.srv,
+		URL:     storeURL(url),
+		Method:  "PUT",
+		Do:      bakeryDo(nil),
+		Header: http.Header{
+			"Content-Type": {"application/json"},
+		},
+		Body: bytes.NewReader(body),
+	}
+	if asAdmin {
+		p.Username = testUsername
+		p.Password = testPassword
+	}
+	httptesting.AssertJSONCall(c, p)
+}
+
+func (s *commonSuite) assertGet(c *gc.C, url string, expectVal interface{}) {
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler:    s.srv,
+		Do:         bakeryDo(nil),
+		URL:        storeURL(url),
+		ExpectBody: expectVal,
+	})
+}
+
+// assertGetUnauthorized asserts that a GET to the given URL results
```

Comment:
wow, that's not what I expected at all reading the tests above. I expected that this was assertGet without any authorization in the request. I think this needs a much clearer name.